```
Error running mint automation
+ mkdir -p 8689-9094bac/gopath/src/github.com/minio
+ git -C 8689-9094bac/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 8689-9094bac/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 8689-9094bac/gopath/src/github.com/minio/minio fetch --quiet minio pull/8689/head:pr8689
+ git -C 8689-9094bac/gopath/src/github.com/minio/minio checkout --quiet pr8689
+ GO111MODULE=on
+ GOPATH=/mnt/nvme/mint-auto/8689-9094bac/gopath
+ make -C 8689-9094bac/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/minio/minio-go/v6 v6.0.45
go: downloading github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/gorilla/mux v1.7.0
go: downloading github.com/json-iterator/go v1.1.7
go: downloading github.com/minio/hdfs/v3 v3.0.1
go: extracting github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/gorilla/mux v1.7.0
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/minio/cli v1.22.0
go: downloading github.com/klauspost/reedsolomon v1.9.3
go: extracting github.com/json-iterator/go v1.1.7
go: extracting github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/dustin/go-humanize v1.0.0
go: downloading github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: extracting github.com/klauspost/reedsolomon v1.9.3
go: downloading github.com/minio/lsync v1.0.1
go: extracting github.com/cheggaaa/pb v1.0.28
go: extracting github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: extracting github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: downloading github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/minio/minio-go/v6 v6.0.45
go: downloading github.com/minio/gokrb5/v7 v7.2.5
go: downloading google.golang.org/api v0.5.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: extracting github.com/minio/lsync v1.0.1
go: extracting github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: downloading github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/klauspost/compress v1.9.7
go: extracting github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/golang/protobuf v1.3.2
go: extracting github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: downloading github.com/djherbis/atime v1.0.0
go: extracting github.com/minio/hdfs/v3 v3.0.1
go: extracting github.com/coreos/etcd v3.3.12+incompatible
go: extracting github.com/djherbis/atime v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.1
go: downloading github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: extracting github.com/mattn/go-colorable v0.1.1
go: downloading github.com/mattn/go-runewidth v0.0.4
go: extracting github.com/golang/protobuf v1.3.2
go: extracting github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: extracting github.com/mattn/go-runewidth v0.0.4
go: downloading github.com/gorilla/handlers v1.4.0
go: extracting github.com/gorilla/handlers v1.4.0
go: downloading github.com/mattn/go-isatty v0.0.7
go: extracting github.com/mattn/go-isatty v0.0.7
go: downloading golang.org/x/crypto v0.0.0-20191117063200-497ca9f6d64f
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/fatih/color v1.7.0
go: extracting github.com/fatih/color v1.7.0
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting golang.org/x/crypto v0.0.0-20191117063200-497ca9f6d64f
go: downloading github.com/prometheus/common v0.2.0
go: extracting github.com/prometheus/common v0.2.0
go: downloading github.com/klauspost/cpuid v1.2.2
go: extracting github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: extracting github.com/klauspost/cpuid v1.2.2
go: downloading github.com/nats-io/stan.go v0.4.5
go: finding github.com/minio/minio/pkg latest
go: finding github.com/minio/minio/pkg/policy latest
go: extracting github.com/nats-io/stan.go v0.4.5
go: downloading google.golang.org/grpc v1.22.0
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting google.golang.org/grpc v1.22.0
go: downloading github.com/Shopify/sarama v1.24.1
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: downloading github.com/gogo/protobuf v1.2.1
go: extracting github.com/Shopify/sarama v1.24.1
go: downloading github.com/lib/pq v1.1.1
go: extracting github.com/lib/pq v1.1.1
go: downloading github.com/go-sql-driver/mysql v1.4.1
go: extracting github.com/go-sql-driver/mysql v1.4.1
go: downloading github.com/coredns/coredns v1.4.0
go: extracting github.com/gogo/protobuf v1.2.1
go: extracting github.com/coredns/coredns v1.4.0
go: downloading gopkg.in/ini.v1 v1.48.0
go: downloading cloud.google.com/go v0.39.0
go: extracting gopkg.in/ini.v1 v1.48.0
go: extracting google.golang.org/api v0.5.0
go: downloading golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading github.com/davecgh/go-spew v1.1.1
go: downloading github.com/beevik/ntp v0.2.0
go: extracting cloud.google.com/go v0.39.0
go: extracting github.com/davecgh/go-spew v1.1.1
go: extracting github.com/beevik/ntp v0.2.0
go: downloading github.com/gorilla/rpc v1.2.0
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.2.3
go: downloading google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: extracting github.com/gorilla/rpc v1.2.0
go: extracting golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: downloading github.com/miekg/dns v1.1.8
go: extracting github.com/miekg/dns v1.1.8
go: extracting google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting gopkg.in/jcmturner/gokrb5.v7 v7.2.3
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/pkg/errors v0.8.1
go: downloading golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: downloading github.com/klauspost/readahead v1.3.1
go: downloading go.opencensus.io v0.21.0
go: extracting github.com/pkg/errors v0.8.1
go: downloading go.uber.org/atomic v1.3.2
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting github.com/klauspost/readahead v1.3.1
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: extracting go.opencensus.io v0.21.0
go: extracting go.uber.org/atomic v1.3.2
go: downloading github.com/shirou/gopsutil v2.18.12+incompatible
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: downloading github.com/eapache/go-resiliency v1.2.0
go: extracting github.com/eapache/go-resiliency v1.2.0
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/shirou/gopsutil v2.18.12+incompatible
go: downloading github.com/hashicorp/vault/api v1.0.4
go: extracting golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/hashicorp/vault/api v1.0.4
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: downloading golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/hashicorp/hcl v1.0.0
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading github.com/modern-go/reflect2 v1.0.1
go: extracting gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/hashicorp/vault/sdk v0.1.13
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/rjeczalik/notify v0.9.2
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/rs/cors v1.6.0
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting github.com/rs/cors v1.6.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: extracting github.com/secure-io/sio-go v0.3.0
go: downloading github.com/minio/sha256-simd v0.1.1
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: extracting github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: extracting github.com/minio/sha256-simd v0.1.1
go: extracting github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: downloading gopkg.in/yaml.v2 v2.2.4
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting gopkg.in/yaml.v2 v2.2.4
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: downloading github.com/minio/parquet-go v0.0.0-20200125064549-a1e49702e174
go: extracting github.com/minio/parquet-go v0.0.0-20200125064549-a1e49702e174
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: extracting github.com/klauspost/pgzip v1.2.1
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: extracting github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/pierrec/lz4 v2.4.0+incompatible
go: extracting golang.org/x/text v0.3.2
go: extracting github.com/minio/gokrb5/v7 v7.2.5
go: extracting github.com/klauspost/compress v1.9.7
go: downloading github.com/fatih/structs v1.1.0
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/ryanuber/go-glob v1.0.0
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting github.com/fatih/structs v1.1.0
go: extracting github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: extracting github.com/ryanuber/go-glob v1.0.0
go: downloading github.com/golang/snappy v0.0.1
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/minio/sio v0.2.0
go: downloading github.com/nats-io/nats.go v1.9.1
go: downloading github.com/tidwall/sjson v1.0.4
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: downloading github.com/tidwall/gjson v1.3.5
go: downloading github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/golang/snappy v0.0.1
go: extracting github.com/minio/sio v0.2.0
go: extracting github.com/tidwall/sjson v1.0.4
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/nats-io/nats.go v1.9.1
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/nats-io/jwt v0.3.2
go: downloading github.com/nats-io/nkeys v0.1.3
go: extracting github.com/tidwall/gjson v1.3.5
go: downloading github.com/tidwall/match v1.0.1
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting github.com/tidwall/pretty v1.0.0
go: extracting github.com/tidwall/match v1.0.1
go: extracting github.com/nats-io/nkeys v0.1.3
go: extracting github.com/nats-io/jwt v0.3.2
go: extracting git.apache.org/thrift.git v0.13.0
go: finding github.com/minio/minio-go/v6 v6.0.45
go: finding google.golang.org/api v0.5.0
go: finding github.com/minio/gokrb5/v7 v7.2.5
go: finding github.com/minio/cli v1.22.0
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/kurin/blazer v0.5.4-0.20190613185654-cf2f27cc0be3
go: finding github.com/Azure/azure-pipeline-go v0.2.1
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/minio/hdfs/v3 v3.0.1
go: finding cloud.google.com/go v0.39.0
go: finding github.com/json-iterator/go v1.1.7
go: finding github.com/gorilla/mux v1.7.0
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/aliyun/aliyun-oss-go-sdk v0.0.0-20190307165228-86c17b95fcd5
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding github.com/mattn/go-colorable v0.1.1
go: finding github.com/Azure/azure-storage-blob-go v0.8.0
go: finding github.com/mattn/go-isatty v0.0.7
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding github.com/fatih/color v1.7.0
go: finding golang.org/x/oauth2 v0.0.0-20190402181905-9f3314589c9a
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding golang.org/x/crypto v0.0.0-20191117063200-497ca9f6d64f
go: finding github.com/rjeczalik/notify v0.9.2
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/mattn/go-ieproxy v0.0.0-20190805055040-f9202b1cfdeb
go: finding google.golang.org/grpc v1.22.0
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding google.golang.org/genproto v0.0.0-20190513181449-d00d292a067c
go: finding golang.org/x/sys v0.0.0-20190922100055-0a153f010e69
go: finding gopkg.in/ini.v1 v1.48.0
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding github.com/secure-io/sio-go v0.3.0
go: finding go.opencensus.io v0.21.0
go: finding golang.org/x/net v0.0.0-20190923162816-aa69164e4478
go: finding github.com/shirou/gopsutil v2.18.12+incompatible
go: finding github.com/golang/protobuf v1.3.2
go: finding github.com/ncw/directio v1.0.5
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding github.com/gogo/protobuf v1.2.1
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/klauspost/readahead v1.3.1
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/coreos/etcd v3.3.12+incompatible
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/klauspost/compress v1.9.7
go: finding github.com/gorilla/handlers v1.4.0
go: finding github.com/klauspost/reedsolomon v1.9.3
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/minio/lsync v1.0.1
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding github.com/beevik/ntp v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding github.com/fatih/structs v1.1.0
go: finding github.com/minio/sio v0.2.0
go: finding github.com/prometheus/client_golang v0.9.3-0.20190127221311-3c4408c8b829
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/miekg/dns v1.1.8
go: finding github.com/skyrings/skyring-common v0.0.0-20160929130248-d1c0bb1cbd5e
go: finding github.com/rs/cors v1.6.0
go: finding gopkg.in/yaml.v2 v2.2.4
go: finding go.uber.org/atomic v1.3.2
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/Shopify/sarama v1.24.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/minio/parquet-go v0.0.0-20200125064549-a1e49702e174
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/nats-io/stan.go v0.4.5
go: finding github.com/hashicorp/vault/api v1.0.4
go: finding golang.org/x/text v0.3.2
go: finding github.com/lib/pq v1.1.1
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/go-sql-driver/mysql v1.4.1
go: finding github.com/nats-io/nats.go v1.9.1
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/mattn/go-runewidth v0.0.4
go: finding github.com/klauspost/cpuid v1.2.2
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/prometheus/client_model v0.0.0-20190129233127-fd36f4220a90
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/common v0.2.0
go: finding github.com/eapache/go-resiliency v1.2.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding gopkg.in/jcmturner/gokrb5.v7 v7.2.3
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/prometheus/procfs v0.0.0-20190117184657-bf6a532e95b1
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/rcrowley/go-metrics v0.0.0-20190704165056-9c2d0518ed81
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/tidwall/sjson v1.0.4
go: finding github.com/tidwall/gjson v1.3.5
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/hashicorp/vault/sdk v0.1.13
go: finding git.apache.org/thrift.git v0.13.0
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.4
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding gopkg.in/square/go-jose.v2 v2.3.1
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding github.com/nats-io/nkeys v0.1.3
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/nats-io/jwt v0.3.2
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
build github.com/minio/minio: cannot load github.com/minio/minio/pkg/policy: no matching versions for query "latest"
make: *** [build] Error 1

```

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -318,7 +318,7 @@ func (a adminAPIHandlers) DataUsageInfoHandler(w http.ResponseWriter, r *http.Re
 
 func (a adminAPIHandlers) AccountingUsageInfoHandler(w http.ResponseWriter, r *http.Request) {
 	ctx := newContext(r, w, "AccountingUsageInfo")
-	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.ListServerInfoAdminAction)
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.AccountingUsageInfoAdminAction)
 	if objectAPI == nil {
 		return
 	}

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -323,7 +323,7 @@ func (a adminAPIHandlers) AccountingUsageInfoHandler(w http.ResponseWriter, r *h
 		return
 	}
 
-	var accountingUsageInfo = make(map[string][]madmin.AccountAccess)
+	var accountingUsageInfo = make(map[string]madmin.BucketAccountingUsage)
 
 	buckets, err := objectAPI.ListBuckets(ctx)
 	if err != nil {
@@ -338,12 +338,25 @@ func (a adminAPIHandlers) AccountingUsageInfoHandler(w http.ResponseWriter, r *h
 		return
 	}
 
+	// Load the latest calculated data usage
+	dataUsageInfo, err := loadDataUsageFromBackend(ctx, objectAPI)
+	if err != nil {
+		logger.LogIf(ctx, err)
+	}
+
 	// Calculate for each bucket, which users are allowed to access to it
 	for _, bucket := range buckets {
+		bucketUsageInfo := madmin.BucketAccountingUsage{}
+
+		// Fetch the data usage of the current bucket
+		if !dataUsageInfo.LastUpdate.IsZero() && dataUsageInfo.BucketsSizes != nil {
+			bucketUsageInfo.Size = dataUsageInfo.BucketsSizes[bucket.Name]
+		}
+
 		for user := range users {
 			rd, wr, custom := globalIAMSys.GetAccountAccess(user, bucket.Name)
 			if rd || wr || custom {
-				accountingUsageInfo[bucket.Name] = append(accountingUsageInfo[bucket.Name], madmin.AccountAccess{
+				bucketUsageInfo.AccessList = append(bucketUsageInfo.AccessList, madmin.AccountAccess{
 					AccountName: user,
 					Read:        rd,
 					Write:       wr,
@@ -351,6 +364,8 @@ func (a adminAPIHandlers) AccountingUsageInfoHandler(w http.ResponseWriter, r *h
 				})
 			}
 		}
+
+		accountingUsageInfo[bucket.Name] = bucketUsageInfo
 	}
 
 	usageInfoJSON, err := json.Marshal(accountingUsageInfo)

--- a/cmd/admin-handlers.go
+++ b/cmd/admin-handlers.go
@@ -316,6 +316,52 @@ func (a adminAPIHandlers) DataUsageInfoHandler(w http.ResponseWriter, r *http.Re
 	writeSuccessResponseJSON(w, dataUsageInfoJSON)
 }
 
+func (a adminAPIHandlers) AccountingUsageInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "AccountingUsageInfo")
+	objectAPI, _ := validateAdminReq(ctx, w, r, iampolicy.ListServerInfoAdminAction)
+	if objectAPI == nil {
+		return
+	}
+
+	var accountingUsageInfo = make(map[string][]madmin.AccountAccess)
+
+	buckets, err := objectAPI.ListBuckets(ctx)
+	if err != nil {
+		// writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrInternalError), r.URL)
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	users, err := globalIAMSys.ListUsers()
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	// Calculate for each bucket, which users are allowed to access to it
+	for _, bucket := range buckets {
+		for user := range users {
+			rd, wr, custom := globalIAMSys.GetAccountAccess(user, bucket.Name)
+			if rd || wr || custom {
+				accountingUsageInfo[bucket.Name] = append(accountingUsageInfo[bucket.Name], madmin.AccountAccess{
+					AccountName: user,
+					Read:        rd,
+					Write:       wr,
+					Custom:      custom,
+				})
+			}
+		}
+	}
+
+	usageInfoJSON, err := json.Marshal(accountingUsageInfo)
+	if err != nil {
+		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+		return
+	}
+
+	writeSuccessResponseJSON(w, usageInfoJSON)
+}
+
 // ServerCPULoadInfo holds informantion about cpu utilization
 // of one minio node. It also reports any errors if encountered
 // while trying to reach this server.

--- a/cmd/admin-router.go
+++ b/cmd/admin-router.go
@@ -56,6 +56,8 @@ func registerAdminRouter(router *mux.Router, enableConfigOps, enableIAMOps bool)
 	// DataUsageInfo operations
 	adminRouter.Methods(http.MethodGet).Path(adminAPIVersionPrefix + "/datausageinfo").HandlerFunc(httpTraceAll(adminAPI.DataUsageInfoHandler))
 
+	adminRouter.Methods(http.MethodGet).Path(adminAPIVersionPrefix + "/accountingusageinfo").HandlerFunc(httpTraceAll(adminAPI.AccountingUsageInfoHandler))
+
 	if globalIsDistXL || globalIsXL {
 		/// Heal operations
 

--- a/cmd/config/policy/opa/config.go
+++ b/cmd/config/policy/opa/config.go
@@ -223,5 +223,6 @@ func (o *Opa) IsAllowed(args iampolicy.Args) (bool, error) {
 		}
 		return resultAllow.Result.Allow, nil
 	}
+
 	return result.Result, nil
 }

--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -28,6 +28,7 @@ import (
 	"github.com/minio/minio/pkg/auth"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
 	"github.com/minio/minio/pkg/madmin"
+	"github.com/minio/minio/pkg/policy"
 )
 
 // UsersSysType - defines the type of users and groups system that is
@@ -1128,6 +1129,109 @@ func (sys *IAMSys) policyDBSet(name, policy string, isSTS, isGroup bool) error {
 		sys.iamGroupPolicyMap[name] = mp
 	}
 	return nil
+}
+
+var iamAccountReadAccessActions = iampolicy.NewActionSet(
+	iampolicy.ListMultipartUploadPartsAction,
+	iampolicy.ListBucketMultipartUploadsAction,
+	iampolicy.ListBucketAction,
+	iampolicy.HeadBucketAction,
+	iampolicy.GetObjectAction,
+	iampolicy.GetBucketLocationAction,
+
+	// iampolicy.ListAllMyBucketsAction,
+)
+
+var iamAccountWriteAccessActions = iampolicy.NewActionSet(
+	iampolicy.AbortMultipartUploadAction,
+	iampolicy.CreateBucketAction,
+	iampolicy.PutObjectAction,
+	iampolicy.DeleteObjectAction,
+	iampolicy.DeleteBucketAction,
+)
+
+var iamAccountOtherAccessActions = iampolicy.NewActionSet(
+	iampolicy.BypassGovernanceModeAction,
+	iampolicy.BypassGovernanceRetentionAction,
+	iampolicy.PutObjectRetentionAction,
+	iampolicy.GetObjectRetentionAction,
+	iampolicy.GetObjectLegalHoldAction,
+	iampolicy.PutObjectLegalHoldAction,
+	iampolicy.GetBucketObjectLockConfigurationAction,
+	iampolicy.PutBucketObjectLockConfigurationAction,
+
+	iampolicy.ListenBucketNotificationAction,
+
+	iampolicy.PutBucketLifecycleAction,
+	iampolicy.GetBucketLifecycleAction,
+
+	iampolicy.PutBucketNotificationAction,
+	iampolicy.GetBucketNotificationAction,
+
+	iampolicy.PutBucketPolicyAction,
+	iampolicy.DeleteBucketPolicyAction,
+	iampolicy.GetBucketPolicyAction,
+)
+
+// GetAccountAccess iterates over all policies documents associated to a user
+// and returns if the user has read and/or write access to any resource.
+func (sys *IAMSys) GetAccountAccess(accountName, bucket string) (rd, wr, o bool) {
+	policies, err := sys.PolicyDBGet(accountName, false)
+	if err != nil {
+		logger.LogIf(context.Background(), err)
+		return false, false, false
+	}
+
+	if len(policies) == 0 {
+		// No policy found.
+		return false, false, false
+	}
+
+	// Policies were found, evaluate all of them.
+	sys.RLock()
+	defer sys.RUnlock()
+
+	var availablePolicies []iampolicy.Policy
+	for _, pname := range policies {
+		p, found := sys.iamPolicyDocsMap[pname]
+		if found {
+			availablePolicies = append(availablePolicies, p)
+		}
+	}
+
+	if len(availablePolicies) == 0 {
+		return false, false, false
+	}
+
+	combinedPolicy := availablePolicies[0]
+	for i := 1; i < len(availablePolicies); i++ {
+		combinedPolicy.Statements = append(combinedPolicy.Statements,
+			availablePolicies[i].Statements...)
+	}
+
+	allActions := iampolicy.NewActionSet(iampolicy.AllActions)
+	for _, st := range combinedPolicy.Statements {
+		// Ignore if this is not an allow policy statement
+		if st.Effect != policy.Allow {
+			continue
+		}
+		// Fast calculation if there is s3:* permissions to any resource
+		if !st.Actions.Intersection(allActions).IsEmpty() {
+			rd, wr, o = true, true, true
+			break
+		}
+		if !st.Actions.Intersection(iamAccountReadAccessActions).IsEmpty() {
+			rd = true
+		}
+		if !st.Actions.Intersection(iamAccountWriteAccessActions).IsEmpty() {
+			wr = true
+		}
+		if !st.Actions.Intersection(iamAccountOtherAccessActions).IsEmpty() {
+			o = true
+		}
+	}
+
+	return
 }
 
 // PolicyDBGet - gets policy set on a user or group. Since a user may

--- a/cmd/iam.go
+++ b/cmd/iam.go
@@ -26,9 +26,9 @@ import (
 	"github.com/minio/minio/cmd/config"
 	"github.com/minio/minio/cmd/logger"
 	"github.com/minio/minio/pkg/auth"
+	"github.com/minio/minio/pkg/bucket/policy"
 	iampolicy "github.com/minio/minio/pkg/iam/policy"
 	"github.com/minio/minio/pkg/madmin"
-	"github.com/minio/minio/pkg/policy"
 )
 
 // UsersSysType - defines the type of users and groups system that is

--- a/pkg/iam/policy/actionset.go
+++ b/pkg/iam/policy/actionset.go
@@ -32,6 +32,11 @@ func (actionSet ActionSet) Add(action Action) {
 	actionSet[action] = struct{}{}
 }
 
+// IsEmpty - returns if the current action set is empty
+func (actionSet ActionSet) IsEmpty() bool {
+	return len(actionSet) == 0
+}
+
 // Match - matches object name with anyone of action pattern in action set.
 func (actionSet ActionSet) Match(action Action) bool {
 	for r := range actionSet {

--- a/pkg/iam/policy/admin-action.go
+++ b/pkg/iam/policy/admin-action.go
@@ -31,6 +31,8 @@ const (
 
 	// StorageInfoAdminAction - allow listing server info
 	StorageInfoAdminAction = "admin:StorageInfo"
+	// AccountingUsageInfoAdminAction - allow listing accounting usage info
+	AccountingUsageInfoAdminAction = "admin:AccountingUsageInfo"
 	// DataUsageInfoAdminAction - allow listing data usage info
 	DataUsageInfoAdminAction = "admin:DataUsageInfo"
 	// PerfInfoAdminAction - allow listing performance info

--- a/pkg/madmin/examples/accounting-usage-info.go
+++ b/pkg/madmin/examples/accounting-usage-info.go
@@ -1,0 +1,45 @@
+// +build ignore
+
+/*
+ * MinIO Cloud Storage, (C) 2019 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package main
+
+import (
+	"log"
+
+	"github.com/minio/minio/pkg/madmin"
+)
+
+func main() {
+	// Note: YOUR-ACCESSKEYID, YOUR-SECRETACCESSKEY and my-bucketname are
+	// dummy values, please replace them with original values.
+
+	// API requests are secure (HTTPS) if secure=true and insecure (HTTPS) otherwise.
+	// New returns an MinIO Admin client object.
+	madmClnt, err := madmin.New("your-minio.example.com:9000", "YOUR-ACCESSKEYID", "YOUR-SECRETACCESSKEY", true)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	accountingUsageInfo, err := madmClnt.AccountingUsageInfo()
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	log.Println(accountingUsageInfo)
+}

--- a/pkg/madmin/info-commands.go
+++ b/pkg/madmin/info-commands.go
@@ -206,9 +206,15 @@ type AccountAccess struct {
 	Custom      bool   `json:"custom"`
 }
 
+// BucketAccountingUsage represents the accounting usage of a particular bucket
+type BucketAccountingUsage struct {
+	Size       uint64          `json:"size"`
+	AccessList []AccountAccess `json:"accessList"`
+}
+
 // AccountingUsageInfo returns the accounting usage info, currently it returns
 // the type of access of different accounts to the different buckets.
-func (adm *AdminClient) AccountingUsageInfo() (map[string][]AccountAccess, error) {
+func (adm *AdminClient) AccountingUsageInfo() (map[string]BucketAccountingUsage, error) {
 	resp, err := adm.executeMethod(http.MethodGet, requestData{relPath: adminAPIPrefix + "/accountingusageinfo"})
 	defer closeResponse(resp)
 	if err != nil {
@@ -221,7 +227,7 @@ func (adm *AdminClient) AccountingUsageInfo() (map[string][]AccountAccess, error
 	}
 
 	// Unmarshal the server's json response
-	var accountingUsageInfo map[string][]AccountAccess
+	var accountingUsageInfo map[string]BucketAccountingUsage
 
 	respBytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {

--- a/pkg/madmin/info-commands.go
+++ b/pkg/madmin/info-commands.go
@@ -198,6 +198,44 @@ func (adm *AdminClient) DataUsageInfo() (DataUsageInfo, error) {
 	return dataUsageInfo, nil
 }
 
+// AccountAccess contains information about
+type AccountAccess struct {
+	AccountName string `json:"accountName"`
+	Read        bool   `json:"read"`
+	Write       bool   `json:"write"`
+	Custom      bool   `json:"custom"`
+}
+
+// AccountingUsageInfo returns the accounting usage info, currently it returns
+// the type of access of different accounts to the different buckets.
+func (adm *AdminClient) AccountingUsageInfo() (map[string][]AccountAccess, error) {
+	resp, err := adm.executeMethod(http.MethodGet, requestData{relPath: adminAPIPrefix + "/accountingusageinfo"})
+	defer closeResponse(resp)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check response http status code
+	if resp.StatusCode != http.StatusOK {
+		return nil, httpRespToErrorResponse(resp)
+	}
+
+	// Unmarshal the server's json response
+	var accountingUsageInfo map[string][]AccountAccess
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(respBytes, &accountingUsageInfo)
+	if err != nil {
+		return nil, err
+	}
+
+	return accountingUsageInfo, nil
+}
+
 // ServerDrivesPerfInfo holds informantion about address and write speed of
 // all drives in a single server node
 type ServerDrivesPerfInfo struct {

